source/robot_model/src/Model.cpp

```diff
@@ -192,10 +192,64 @@ StateRepresentation::CartesianPose Model::forward_geometry(const StateRepresenta
   return this->forward_geometry(joint_state, std::vector<std::string>{frame_name}).front();
 }
 
-StateRepresentation::JointPositions Model::inverse_geometry(const StateRepresentation::CartesianState&) const {
-  // TODO
-  return StateRepresentation::JointPositions();
-}
+StateRepresentation::JointPositions Model::inverse_geometry(StateRepresentation::JointState actual_joint_state,
+                                                              const StateRepresentation::CartesianState &desired_cartesian_state) {
+    // This method uses the angular velocity evaluated by the 'Infeasibility-free inverse kinematics method' which takes into account the
+    // joint limits. The angular velocity is then used for indicating the direction for an iterative algorithm.
+    // To converge at the desired end-effector position a similar approach to the Newton–Raphson method is used, the only difference is that
+    // here the pseudo-inverse of the jacobian is not used and instead is used the angualr velocity evaluated by the
+    // 'Infeasibility-free inverse kinematics method'.
+
+    const int JOINT_ID = this->robot_model_.njoints - 2;
+
+    // TODO if the actual_joint_state is not provided initialize the position of the robot in a radom conformation
+    // TODO take as input the parameters eps and IT_MAX
+    // TODO return 'success'
+
+    Eigen::VectorXd q = actual_joint_state.get_positions(); //pinocchio::neutral(model);
```

Comment:
it was because I could need a generalized robot position if it is not provided as input in the function, but it is better to have a random one and the `neutral`, because it sets to zero all the joint angles.
And yes I forgot to remove the comment.

Reply:
done!